index.js

```diff
@@ -571,7 +572,7 @@ module.exports = class Hypercore extends EventEmitter {
       }
     }
 
-    return await this.core.append(buffers, this.sign, { preappend })
+    return await this.core.append(buffers, this.auth.sign, { preappend })
```

Comment:
let's just pass this.auth down instead and change the interface in auth. means it can be an class instance also which is nice